### ISSUE TITLE
Variant for use in browsers

### ISSUE THREAD

**Issue by adriancole** [openzipkin/zipkin-js#1]
From @virusdave

>  I want to get a whole-system tracing tree that is rooted in a client's action on a web page. While there is a zipkin-js project, this appears to be a server-side node.js library. Is there a canonical client-side JS library that one would use to do things like 'start a new trace', 'annotate trace with browser info', 'send trace data to a specific collector', etc?

**Comment by eirslett** [comment]
I think we should be able to refactor the code base a little bit, so you can pass in a context API as a constructor argument instead of hard-coding to CLS. That would make it more universal. We could e.g. use AngularJS Zones for the browser part.

We need to build a zipkin-transport-http first, since I think http is the only viable option for sending trace data from the browser.

There are also some other problems, like how to protect zipkin-server from potentially malicious requests, if it's exposed to the wide Internet?

**Comment by adriancole** [comment]
>
> There are also some other problems, like how to protect zipkin-server from
> potentially malicious requests, if it's exposed to the wide Internet?
>
good point in general, though it isn't necessarily something we need to
address at the same time since there's a use case for tracing internal or
otherwise vpn'd apps. wanna open an issue in zipkin-java? we can start
collecting practice.
